### ISSUE TITLE
On-canvas transform and anchor points UI flashes/flickers adjusting a Rect's radius

### ISSUE THREAD

**Issue by chadrolfs** [adobe-photoshop/spaces-design#3014]
cmd/ctrl + n
r for rect tool
drag a rect shape on canvas
adjust Radius slider

Result: on canvas UI flickers.

**Comment by chadrolfs** [comment]
happens on mac and win

**Comment by chadrolfs** [comment]
does not happen when superselect tool is active unless you are in vector edit mode.

**Comment by iwehrman** [comment]
It looks like we're getting historyStateNotifications when we shouldn't be? Here's what I'm seeing: 
```
Enqueuing action transform.setRadius; 0/0
fluxcontroller.js:662 Executing action transform.setRadius after waiting 0ms; 1/0
echo-play.js:25 batchPlay: ([
 {
  "name": "changePathDetails",
  "descriptor": {
   "keyActionChangeAllCorners": true,
   "keyOriginRRectRadii": {
    "_obj": "radii",
    "_value": {
     "topLeft": {
      "_unit": "pixelsUnit",
      "_value": 36
     },
     "topRight": {
      "_unit": "pixelsUnit",
      "_value": 36
     },
     "bottomLeft": {
      "_unit": "pixelsUnit",
      "_value": 36
     },
     "bottomRight": {
      "_unit": "pixelsUnit",
      "_value": 36
     },
     "unitValueQuadVersion": 1
    }
   },
   "keyOriginType": 1
  },
  "options": {
   "interactionMode": 3
  }
 }
], {
 "paintOptions": {
  "immediateUpdate": true,
  "quality": "draft"
 },
 "historyStateInfo": {
  "name": "Set Radius",
  "target": {
   "_ref": "document",
   "_id": 1115
  },
  "coalesce": true,
  "suppressHistoryStateNotification": true
 }
}, undefined);
history.js:644 Updating the previous history state instead of pushing (coalesce or rogue-catchup)
history.js:664 History state added or merged: 3
all-events.js:11 ('historyState', {
 "ID": 1142,
 "currentHistoryState": 3,
 "documentID": 1115,
 "historyStates": 3,
 "name": "Set Radius"
});
history.js:305 History state event from photoshop (raw): currentState (index) 3, total states: 3
fluxcontroller.js:641 Enqueuing action history.handleHistoryState; 1/0
echo-play.js:27 batchPlay result: [
  {}
]
fluxcontroller.js:672 Finished action transform.setRadius in 70ms with RTT 70ms; 1/1
fluxcontroller.js:662 Executing action history.handleHistoryState after waiting 2ms; 1/0
fluxcontroller.js:672 Finished action history.handleHistoryState in 2ms with RTT 4ms; 1/0
fluxcontroller.js:641 Enqueuing action transform.setRadius; 0/0
fluxcontroller.js:662 Executing action transform.setRadius after waiting 1ms; 1/0
echo-play.js:25 batchPlay: ([
 {
  "name": "changePathDetails",
  "descriptor": {
   "keyActionChangeAllCorners": true,
   "keyOriginRRectRadii": {
    "_obj": "radii",
    "_value": {
     "topLeft": {
      "_unit": "pixelsUnit",
      "_value": 41
     },
     "topRight": {
      "_unit": "pixelsUnit",
      "_value": 41
     },
     "bottomLeft": {
      "_unit": "pixelsUnit",
      "_value": 41
     },
     "bottomRight": {
      "_unit": "pixelsUnit",
      "_value": 41
     },
     "unitValueQuadVersion": 1
    }
   },
   "keyOriginType": 1
  },
  "options": {
   "interactionMode": 3
  }
 }
], {
 "paintOptions": {
  "immediateUpdate": true,
  "quality": "draft"
 },
 "historyStateInfo": {
  "name": "Set Radius",
  "target": {
   "_ref": "document",
   "_id": 1115
  },
  "coalesce": true,
  "suppressHistoryStateNotification": true
 }
}, undefined);
history.js:644 Updating the previous history state instead of pushing (coalesce or rogue-catchup)
history.js:664 History state added or merged: 3
all-events.js:11 ('historyState', {
 "ID": 1144,
 "currentHistoryState": 3,
 "documentID": 1115,
 "historyStates": 3,
 "name": "Set Radius"
});
history.js:305 History state event from photoshop (raw): currentState (index) 3, total states: 3
fluxcontroller.js:641 Enqueuing action history.handleHistoryState; 1/0
echo-play.js:27 batchPlay result: [
  {}
]
fluxcontroller.js:672 Finished action transform.setRadius in 61ms with RTT 62ms; 1/1
fluxcontroller.js:662 Executing action history.handleHistoryState after waiting 3ms; 1/0
fluxcontroller.js:672 Finished action history.handleHistoryState in 1ms with RTT 4ms; 1/0
fluxcontroller.js:641 Enqueuing action transform.setRadius; 0/0
fluxcontroller.js:662 Executing action transform.setRadius after waiting 0ms; 1/0
echo-play.js:25 batchPlay: ([
 {
  "name": "changePathDetails",
  "descriptor": {
   "keyActionChangeAllCorners": true,
   "keyOriginRRectRadii": {
    "_obj": "radii",
    "_value": {
     "topLeft": {
      "_unit": "pixelsUnit",
      "_value": 42
     },
     "topRight": {
      "_unit": "pixelsUnit",
      "_value": 42
     },
     "bottomLeft": {
      "_unit": "pixelsUnit",
      "_value": 42
     },
     "bottomRight": {
      "_unit": "pixelsUnit",
      "_value": 42
     },
     "unitValueQuadVersion": 1
    }
   },
   "keyOriginType": 1
  },
  "options": {
   "interactionMode": 3
  }
 }
], {
 "paintOptions": {
  "immediateUpdate": true,
  "quality": "draft"
 },
 "historyStateInfo": {
  "name": "Set Radius",
  "target": {
   "_ref": "document",
   "_id": 1115
  },
  "coalesce": true,
  "suppressHistoryStateNotification": true
 }
}, undefined);
history.js:644 Updating the previous history state instead of pushing (coalesce or rogue-catchup)
history.js:664 History state added or merged: 3
all-events.js:11 ('historyState', {
 "ID": 1146,
 "currentHistoryState": 3,
 "documentID": 1115,
 "historyStates": 3,
 "name": "Set Radius"
});
history.js:305 History state event from photoshop (raw): currentState (index) 3, total states: 3
fluxcontroller.js:641 Enqueuing action history.handleHistoryState; 1/0
echo-play.js:27 batchPlay result: [
  {}
]
fluxcontroller.js:672 Finished action transform.setRadius in 59ms with RTT 59ms; 1/1
```

**Comment by jsbache** [comment]
CL: 1048932 implements a core change where we ensure that overlay views are updated before we perform an immediate draw

**Comment by iwehrman** [comment]
Over to @chadrolfs to verify.

**Comment by jsbache** [comment]
Fix hass been promoted to staging in CL: 1049124
